### DISCUSSION TITLE
help wanted: delegate authentication to an external IdP (OIDC/OAuth2)

### DISCUSSION THREAD

**Discussion by jyio** [open-webui/open-webui#340] [General]
Thank you for sharing this awesome UI. Happy new year!

#68 adds a very useful feature, and I'd like to integrate it with my existing auth stack. How would one delegate authentication to an external IdP (for example, OIDC/OAuth2 or forward auth headers)?

**Comment by justinh-rahb** [comment]
I'm hopeful this is something they already had in mind and I suspect so from looking at the discussion of that feature merge. There's still much to do before I could expose this to the Internet with confidence, but all in good time.

**Comment by tjbck** [comment]
Hey guys! Happy New Year! 

I'd love to implement a feature that would delegate authentication to external services, but I'm trying to focus on other core features like RAG, function calling, etc. at the moment, so I'm not entirely sure when I'll be able to take a look at this. If anyone could get the ball rolling for this feature and create a PR, I'd greatly appreciate it. Thanks!

**Comment by jyio** [comment]
Totally understandable that there are other priorities. I'll try to look into it when I get a chance. Thank you for considering it :)

**Comment by Collected5353** [comment]
Shouldn't be too hard since https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/ already contains most of the code I can take look to see what we would need to implement. 

I think handling public Oauth should be straightforward. Configuring external IDP may take more time and testing.

**Reply by justinh-rahb** [reply]
Working with Google, Microsoft etc. yes should be fairly easy. I feel this would probably cover most people that want OIDC for a start. Generic IDPs such as KeyCloak, Authentik, Authelia, etc. are more difficult, because you'll need to stand up that infrastructure for testing.

**Comment by Collected5353** [comment]
Right now the best advice I can give is to run ngrok to integrate Oauth. Got it working for google with no issues.

**Comment by sammcj** [comment]
Has anyone got an example of a redirect URL / request that passes in a JWT for authentication?

I want to bypass the internal authentication page as I'm using Authentik as my IDP across all apps.

**Comment by UberMetroid** [comment]
I am ready and willing to test out LDAP once it gets added

**Comment by yousecjoe** [comment]
I'm interested in testing this for Zero Trust IAM requirements using Keycloak, Entra, JumpCloud, and others.

**Comment by Yanyutin753** [comment]
[Please try clerk](https://clerk.com/)

**Comment by longregen** [comment]
I believe this was done with https://github.com/open-webui/open-webui/pull/2574, unless you are referring to the actual requests made by agents
